auth_proxy/models/oauth.py

```diff
@@ -21,8 +21,11 @@ class Client(Base):
     client_id = Column(String, primary_key=True)
     client_secret = Column(String, nullable=False)
 
-    _redirect_uris = Column(Text)
-    _default_scopes = Column(Text)
+    user_id = Column(Integer, ForeignKey('user.id'))
+    user = orm.relationship('User')
```

Comment:
Okay, yeah. The way I'm understanding it, users and clients probably shouldn't be connected directly to each other at all, they only connection should be through the Token (authorization).

![office lens 20160603-133524](https://cloud.githubusercontent.com/assets/502000/15787392/2a1dfc42-2990-11e6-916e-c0eee8441eaa.jpg)